k2/csrc/ragged.cu

```diff
@@ -208,11 +213,20 @@ RaggedShape RaggedShape::To(ContextPtr ctx) const {
   return RaggedShape(axes);
 }
 
+RaggedShape RaggedShape::Clone() const {
+  RaggedShape ans;
+  ans.axes_.reserve(axes_.size());
+  const_cast<RaggedShape *>(this)->Populate();
```

Comment:
If src has no valid row_ids, maybe it's better to let dest has no valid row_ids either? User can populate dest if they want.

Reply:
Agreed, it's not necessary to populate here, I think.
(Although you could make an argument that it might save some duplicate work later).